### PR TITLE
Introduce CompositeUserRepository #196

### PR DESCRIPTION
**Pull request by glebshalyganov**
Introduce CompositeUserRepository #196

### CODE REVIEW

File: core/src/main/java/io/jmix/core/security/CompositeUserRepository.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        for (UserRepository delegate : userRepositories) {
+            UserDetails userDetails;
+            try {
+                userDetails = delegate.loadUserByUsername(username);
+            } catch (Exception e) {
```
Comment (andreysubbotin) [inline core/src/main/java/io/jmix/core/security/CompositeUserRepository.java:64]: Perhaps it is worth catching a `UsernameNotFoundException`?

File: core/src/main/java/io/jmix/core/security/CompositeUserRepository.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    protected List<UserRepository> userRepositories;
+
+    @Override
+    public UserDetails getSystemUser() {
+        for (UserRepository delegate : userRepositories) {
+            UserDetails systemUser;
+            try {
+                systemUser = delegate.getSystemUser();
+            } catch (Exception e) {
+                continue;
```
Comment (andreysubbotin) [inline core/src/main/java/io/jmix/core/security/CompositeUserRepository.java:30]: Log the error message

File: core/src/main/java/io/jmix/core/security/CompositeUserRepository.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    public UserDetails getAnonymousUser() {
+        for (UserRepository delegate : userRepositories) {
+            UserDetails anonymousUser;
+            try {
+                anonymousUser = delegate.getAnonymousUser();
+            } catch (Exception e) {
+                continue;
```
Comment (andreysubbotin) [inline core/src/main/java/io/jmix/core/security/CompositeUserRepository.java:44]: Log the error message

**Review by andreysubbotin — CHANGES_REQUESTED** [review]

**Comment by glebshalyganov** [comment]
merged
https://github.com/Haulmont/jmix-core/commit/9627bdec9c115966a64f163cef02bf64d13ca40e
https://github.com/Haulmont/jmix-core/commit/4e74c686d499f122268a8321cc841b371f27efb9

**Pull request closed without merging**